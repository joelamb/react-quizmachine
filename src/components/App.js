import React from 'react';
import OptionsContainer from '../containers/OptionsContainer';
import QuestionContainer from '../containers/QuestionContainer';
import ScoreBoardContainer from '../containers/ScoreboardContainer';
import '../styles/app.scss';
import { fetchQuestion } from '../actions';

class App extends React.Component {

  componentDidMount() {
    fetchQuestion();
  }

  render() {
    return (
      <main>
        <h1>App contents go here</h1>
        <OptionsContainer />
        <QuestionContainer />
        <ScoreBoardContainer />
      </main>
    )
  }
}

export default App;
