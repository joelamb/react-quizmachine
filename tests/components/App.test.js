import React from 'react'
import App from '../../src/components/App';
import renderer from 'react-test-renderer';

describe('App', () => {
  test('should render correctly', () => {
    const mockFetchHiScores = jest.fn();
    const mockFetchQuestion = jest.fn();
    const tree = renderer
      .create(<App fetchHiScores={mockFetchHiScores} fetchQuestion={mockFetchQuestion} />)
      .toJSON()
    expect(tree).toMatchSnapshot();
  });
})