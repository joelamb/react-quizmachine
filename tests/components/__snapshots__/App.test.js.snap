// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<main>
  <header>
    <div>
      <button
        onClick={[Function]}
      >
        <i
          className="material-icons"
        >
          menu
        </i>
      </button>
      <h1>
        Mastermind
      </h1>
    </div>
  </header>
  <section
    className="page"
  />
</main>
`;
